inference-engine/src/mkldnn_plugin/nodes/psroi.cpp

```diff
@@ -42,250 +46,418 @@ class PSROIPoolingImpl: public ExtLayerBase {
+                roiEndW   = static_cast<float>(round(bottomRois[3] + 1.0f)) * spatialScale;
+                roiEndH   = static_cast<float>(round(bottomRois[4] + 1.0f)) * spatialScale;
+                // Force too small ROIs to be 1x1
+                roiWidth  = std::max<float>(roiEndW - roiStartW, 0.1f);  // avoid 0
+                roiHeight = std::max<float>(roiEndH - roiStartH, 0.1f);
+                if (inFmt == Layout::NHWC) {
+                    parallel_for2d(nh, nw, [&](int h, int w) {
+                        for (int c = 0; c < nc; c++) {
+                            const int gc = (c * groupSize + h) * groupSize + w;
+                            const int binOffsetInput = roiBatchInd * channels * height * width + gc;
+                            const int binOffsetOutput = n * nc * nh * nw + c;
+                            avgPsroi(c, h, w, 0, 0, binOffsetInput, binOffsetOutput);
+                        }
+                    });
+                } else {  // nchw, nChw16c, nChw8c
+                    parallel_for3d(outBlockCount, nh, nw, [&](int blkIdx, int h, int w) {
+                        int cStart = blkIdx * outBlockSize;
+                        int cEnd = (blkIdx == outBlockCount - 1 ? nc : cStart + outBlockSize);
+                        for (int c = cStart; c < cEnd; c++) {
+                            const int gc = (c * groupSize + h) * groupSize + w;
+                            const int inputBlockResidual = (inFmt == Layout::NCHW ? 0 : gc % inBlockSize);
+                            const int outputBlockResidual = (outFmt == Layout::NCHW ? 0 : c % inBlockSize);
+                            const int inputBlockIdx = (gc / inBlockSize) * inBlockSize;
+                            const int outputBlockIdx = (c / outBlockSize) * outBlockSize;
+                            const int binOffsetInput = (roiBatchInd * inputChannelsPadding + inputBlockIdx) * height * width;
+                            const int binOffsetOutput = (n * outputChannelsPadding + outputBlockIdx) * nh * nw;
+                            avgPsroi(c, h, w, inputBlockResidual, outputBlockResidual, binOffsetInput, binOffsetOutput);
+                        }
+                    });
+                }
+            } else if (mode == "bilinear") {
+                roiStartW = bottomRois[1] * spatialScale;
+                roiStartH = bottomRois[2] * spatialScale;
+                roiEndW = bottomRois[3] * spatialScale;
+                roiEndH = bottomRois[4] * spatialScale;
+                roiWidth  = roiEndW - roiStartW;
+                roiHeight = roiEndH - roiStartH;
+
+                int outputBlockIdx, binOffsetOutput, outputBlockResidual;
+                if (inFmt == Layout::NHWC) {
+                    parallel_for2d(nh, nw, [&](int h, int w) {
+                        for (int c = 0; c < nc; c++) {
+                            binOffsetOutput = n * nc * nh * nw + c;
```

Comment:
`n * nc * nh * nw` could be calculated outside the "nc" loop.

Reply:
Done